third_party/amd/lib/TritonAMDGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -555,6 +555,76 @@ struct BufferStoreOpConversion
   }
 };
 
+struct BufferAtomicFAddOpConversion
+    : public ConvertOpToLLVMPattern<triton::amdgpu::BufferAtomicFAddOp>,
+      public LoadStoreConversionBase {
+  using ConvertOpToLLVMPattern<
+      triton::amdgpu::BufferAtomicFAddOp>::ConvertOpToLLVMPattern;
+
+  BufferAtomicFAddOpConversion(LLVMTypeConverter &converter,
+                               const AMD::TargetInfo &targetInfo,
+                               ModuleAxisInfoAnalysis &axisAnalysisPass,
+                               PatternBenefit benefit)
+      : ConvertOpToLLVMPattern<triton::amdgpu::BufferAtomicFAddOp>(converter,
+                                                                   benefit),
+        LoadStoreConversionBase(targetInfo, axisAnalysisPass) {}
+
+  LogicalResult
+  matchAndRewrite(triton::amdgpu::BufferAtomicFAddOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto loc = op->getLoc();
+    LLVM::AMD::BufferEmitter bufferEmitter(rewriter, loc, targetInfo);
+
+    // original values
+    Value ptr = op.getPtr();
+    Value offset = op.getOffsets();
+    Value mask = op.getMask();
+    Value data = op.getValue();
+
+    Value llPtr = adaptor.getPtr();
+    Value llOffset = adaptor.getOffsets();
+    Value llMask = adaptor.getMask();
+    Value llData = adaptor.getValue();
+
+    // Determine the vectorization size
+    Type valueTy = data.getType();
+    Type valueElemTy =
+        typeConverter->convertType(getElementTypeOrSelf(valueTy));
+    Type ptrType = getPointerTypeWithShape(ptr, offset);
+
+    unsigned numElems = getTotalElemsPerThread(ptrType);
+    assert((valueElemTy.isBF16() && numElems == 2) ||
+           (valueElemTy.isF16() && numElems == 2) ||
+           (valueElemTy.isF32() && numElems == 1) &&
+               "buffer_atomic_fadd only supports bf16|f16|f32 element type and "
+               "2|2|1 "
+               "num elements per thread");
```

Comment:
Also yes this is basically a copy/paste of `buffer_store` which I will refactor to be simpler as soon as we agree this is the correct path.

Reply:
Yes, I think as a first draft you can start by disallowing 1xf16 cases. But Ilia has a trick to do dpp swizzling, so that each other thread gets those two elements and then you can still get issue this instruction.